pytradfri/smart_task.py

```diff
@@ -145,34 +191,38 @@ def __repr__(self):
 class TaskControl:
     """Class to control the tasks."""
 
-    def __init__(self, task, state: bool, path, gateway):
+    def __init__(
+        self, task: SmartTask, state: bool, path: list[str], gateway: Gateway
+    ) -> None:
         """Initialize TaskControl."""
         self._task = task
         self.state = state
         self.path = path
         self._gateway = gateway
 
     @property
-    def tasks(self):
+    def tasks(self) -> list[StartActionItem]:
         """Return task objects of the task control."""
         return [
-            StartActionItem(self._task, i, self.state, self.path, self.raw)
-            for i in range(len(self.raw))
+            StartActionItem(self._task, idx, self.state, self.path, self.raw)
+            for idx in range(len(self.raw.root_start_action))
         ]
 
-    def set_dimmer_start_time(self, hour, minute):
+    def set_dimmer_start_time(self, hour: int, minute: int) -> Command[None]:
         """Set start time for task (hh:mm) in iso8601.
 
         NB: dimmer starts 30 mins before time in app
         """
         #  This is to calculate the difference between local time
         #  and the time in the gateway
-        d_now = self._gateway.get_gateway_info().current_time
+        gateway_info: Command[GatewayInfo] = self._gateway.get_gateway_info()
+        d_now = gateway_info.current_time  # type: ignore[attr-defined]
```

Comment:
This is a bug. We need to execute the command with an api before we can access the result.

Reply:
Hm, do we really want an API call here? One of the features of this library is to not have API calls in the code representing the gateway's responses?

Reply:
(I understand that it's a bug)

Reply:
Perhaps a way forward is to assume the gateway has the same local time as the environment where the script is run?

Reply:
Maybe add a method, eg `calibrate_time`, that returns a Command with a result callback that sets the time diff between local time and gateway time as an attribute on the smart task?

We default this attribute to timedelta(0).

Reply:
Tried implementing something along the line of your comments here